exec/java-exec/src/main/java/org/apache/drill/exec/ops/OperatorContextImpl.java

```diff
@@ -130,6 +148,27 @@ public OperatorStats getStats() {
     return stats;
   }
 
+  public <RESULT> ListenableFuture<RESULT> runCallableAs(final UserGroupInformation proxyUgi,
+                                                         final Callable<RESULT> callable) {
+    synchronized (this) {
```

Comment:
I think we should use the WorkManager pool for this rather than creating another thread pool.